central/image/datastore/datastore_impl.go

```diff
@@ -303,23 +302,15 @@ func (ds *datastoreImpl) initializeRankers() {
 		sac.AllowFixedScopes(
 			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS), sac.ResourceScopeKeys(resources.Image)))
 
-	results, err := ds.searcher.Search(readCtx, pkgSearch.EmptyQuery())
+	// The bespoke store does not have a Walk nor a GetByQueryFn.  Thus, WalkByQuery must be used.
+	err := ds.storage.WalkByQuery(readCtx, pkgSearch.EmptyQuery(), func(image *storage.Image) error {
+		ds.imageRanker.Add(image.GetId(), image.GetRiskScore())
```

Comment:
Add is synchronized and recomputes ranks on every call
https://github.com/stackrox/stackrox/blob/815cc9c1bace8e904dee0d2ec57198633d9bcb26/central/ranking/ranker.go#L56-L59

How about loading data in batch to prevent multiple locking and recomputing? So instead of calling `Add` create a map and after `Walk`ing call `BulkAdd`.
But this will deserve a dedicated PR

Reply:
That is a good idea for 4.9 or later.  Though we are also going to be having conversations about whether this ranker concept still makes sense in general as its primary use is in deprecated dashboards.